src/ext/java/org/opentripplanner/ext/fares/model/FareLegRule.java

```diff
@@ -8,6 +8,9 @@ public record FareLegRule(
   @Nullable String networkId,
   @Nullable String fromAreaId,
   @Nullable String toAreadId,
+  @Nullable Double minDistance,
```

Comment:
How should we deal with the number of stops metric in this case? It seems difficult to put both distance types into a single data type in a nice way

Reply:
Ah, I see. So it can be either number of stops or linear distance, but not both, right?

I think I would introduce polymorphic type which encapsulates both the type and the min/nax values. Something like this:

```java
sealed interface FareDistance {
  record Stops(int min, int max) implements FareDistance {};
  record LinearDistance(Distance min, Distance max) implements FareDistance {};
}
```

In Java 21, which will be released later this year, we could then write the following:

```
return switch (o) {
     case Stops stops -> ... // logic for stops
     case LinearDistance distance -> // logic for distance
};
```    

In the meantime we will have to make do with an instanceof check:

```
if(o instance of Stops stops){
  // logic for stops
}
else if (o instance of LinearDistance ld) {
  // logic for distance
}
```

Reply:
Seems like a good idea. Would we still need a DistanceType enum to represent the different types of distance if this is implemented? It seems to me like it would be unnecessary to have a distance type property inside a Stops or LinearDistance instance

Reply:
No, no enum is necessary. That one class would replace all 3 fields.